src/cephadm/cephadmlib/systemd_unit.py

```diff
@@ -216,9 +221,12 @@ def update_files(
     *,
     init_container_ids: Optional[List[DaemonSubIdentity]] = None,
     sidecar_ids: Optional[List[DaemonSubIdentity]] = None,
+    limit_core_infinity: Optional[bool] = False,
```

Comment:
Since this value seems to *always* be taken from ctx, you probably could just use `getattr(ctx, 'limit_core_infinity', None)` inside the function. Optionally having a new extra argument for one-off overrides. However, it's not a big deal